### ISSUE TITLE
Circles appear behind other layers

### ISSUE THREAD

**Issue by ryanhamley** [smithmicro/mapbox-gl-circle#50]
Hi, I started implementing mapbox-gl-circle into a program that has several other layers on the map and noticed that the circle always appeared behind one of my other layers, regardless of what order they were added in.

I tracked it down to [these lines](https://github.com/mblomdahl/mapbox-gl-circle/blob/master/lib/main.js#L940-L941) in the code where `waterway-label` is hardcoded as the second argument to `addLayer` causing the circle to always be added just before this layer.

You can see the issue in action here:
<img width="745" alt="screen shot 2017-10-10 at 11 20 58 am" src="https://user-images.githubusercontent.com/4523080/31403351-b5949f86-adad-11e7-96b4-c575a37ba516.png">

It's pretty straightforward to expose the `before` argument to the user so that users can control where a circle is added in the layer stack. I implemented a fix as you can see in this image.

<img width="864" alt="screen shot 2017-10-10 at 11 16 46 am" src="https://user-images.githubusercontent.com/4523080/31403350-b580dc4e-adad-11e7-9b73-0de2f073ca06.png">

Here's my pull request for the proposed fix https://github.com/mblomdahl/mapbox-gl-circle/pull/51

Thanks!

**Comment by mblomdahl** [comment]
@ryanhamley I noticed a bug with this implementation; 

![screen shot 2017-10-22 at 4 49 52 pm](https://user-images.githubusercontent.com/786326/31865278-bd270dba-b76c-11e7-9b23-150afcd29c55.png)

Notice that the handle in the center is now getting blurred by overlapping circles. Handles must always be at the top, so I modified the code in 09a19f4 to use 'waterway-label' by default for fill/stroke, but still allows you to force the layers to end of layers array by doing `myCircle.addTo(myMap, null)`. Released to NPM with #54.
